### PR TITLE
allow flutter tester to disable font loading from asset bundle

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards https://github.com/flutter/flutter/issues/103724

TLDR: We want to make the tester able to load assets by passing flutter-asset-dir. Unfortunately that also causes tests to fallback to bundled fonts ahead of Ahem font which is massively breaking to scuba and not necessarily something we want to be easy. Add a flag that prevents registering fonts located in the asset bundle.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I've added tests that assert that with the flag passed a bundled roboto font is not used (verified this test fails with the flag removed). I also added a test that we can still add new fonts by using the loadFont API

File: shell/common/switches.h [hunk 1]
```diff
@@ -176,6 +176,11 @@ DEF_SWITCH(UseTestFonts,
            "will make font resolution default to the Ahem test font on all "
            "platforms (See https://www.w3.org/Style/CSS/Test/Fonts/Ahem/). "
            "This option is only available on the desktop test shells.")
+DEF_SWITCH(DisableAssetFonts,
+           "disable-asset-fonts",
+           "Prevents usage of any non-test fonts unless they were explicitly "
+           "Loaded via dart:ui font APIs. This option is only available on the "
```
Comment (zanderso) [inline shell/common/switches.h:182]: How is it kept only available in the test shell?

Comment (jonahwilliams) [reply shell/common/switches.h:182]: oh right. hmm. I think should probably ifdef this out

Comment (jonahwilliams) [reply shell/common/switches.h:182]: well, I guess the thinking is that these switches aren't on an allowlist or surfaced through the embedder, but this one and use-test-fonts are technically still there. We should fix that

Comment (jonahwilliams) [reply shell/common/switches.h:182]: ahh, but if we change a define for switches then we'd have to compile this twice for each host platform

🤔

Comment (jonahwilliams) [reply shell/common/switches.h:182]: Added a debug mode guard

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**